packages/app/hooks/use-wallet/use-wallet.ts

```diff
@@ -20,37 +22,58 @@ const useWallet = (): UseWalletReturnType => {
         setAddress(getAddress(connector.session.accounts[0]));
       } else if (magicWalletAddress) {
         setAddress(magicWalletAddress);
+      } else if (mobileSDK.address) {
+        setAddress(mobileSDK.address);
       } else if (web3) {
         const address = await web3.getSigner().getAddress();
         setAddress(address);
       } else {
         setAddress(undefined);
       }
     })();
-  }, [web3, connector.session, magicWalletAddress]);
+  }, [web3, connector.session, magicWalletAddress, mobileSDK.address]);
 
   const result = useMemo(() => {
+    const wcConnected = connector.connected;
+    const mobileSDKConnected = mobileSDK.connected;
+
+    let walletName: string | undefined;
+    if (wcConnected) {
+      walletName = connector.peerMeta?.name;
+    } else if (mobileSDKConnected) {
+      walletName = mobileSDK.metadata?.name;
+    }
+
     return {
       address,
       connect: async () => {
-        await connector.connect();
+        await Promise.race([connector.connect(), mobileSDK.onConnected()]);
```

Comment:
Had to get a bit creative here since the fullscreen modal showing the list of wallets is quite coupled to WalletConnect. Open to other suggestions.

1. Calling `connector.connect()` here is needed to display the wallet list modal. 
2. When the "Coinbase Wallet" option is pressed it initiates the handshake process with CBW
3. After a successful handshake, the `mobileSDK.onConnected()` resolves this promise and sign in process can continue

Reply:
I think this was genius 💯